virtual_rainforest/core/data.py

```diff
@@ -0,0 +1,398 @@
+        used in the Data instance. That routine is used to validate the DataArray and
+        then add the DataArray to the Data dictionary or replace the existing DataArray
+        under that key.
+
+        Note that the DataArray name is expected to match the standard internal variable
+        names used in Virtual Rainforest.
+
+        Args:
+            key: The name to store the data under
+            value: The DataArray to be stored
+        """
+
+        if not isinstance(value, DataArray):
+            log_and_raise(
+                "Only DataArray objects can be added to Data instances", TypeError
+            )
+
+        # Ensure dataarray name matches the key
+        value.name = key
+
+        if key not in self.data.data_vars:
+            LOGGER.info(f"Adding data array for '{key}'")
+        else:
+            LOGGER.info(f"Replacing data array for '{key}'")
+
+        # Validate and store the data array
+        value, valid_dict = validate_dataarray(value=value, grid=self.grid)
+        self.data[key] = value
+        self.variable_validation[key] = valid_dict
+
+    def __getitem__(self, key: str) -> DataArray:
+        """Get a given data variable from a Data instance.
+
+        This method looks for the provided key in the data variables saved in the `data`
+        attribute and returns the DataArray for that variable. Note that this is just a
+        shortcut: ``data_instance['var']`` is the same as ``data_instance.data['var']``.
+
+        Args:
+            key: The name of the data variable to get
+
+        Raises:
+            KeyError: if the data variable is not present
+        """
+
+        return self.data[key]
+
+    def __contains__(self, key: str) -> bool:
+        """Check if a given data variable is present in a Data instance.
+
+        This method provides the `var_name in data_instance` functionality for a Data
+        instance. This is just a shortcut: ``var in data_instance`` is the same as
+        ``var in data_instance.data``.
+
+        Args:
+            key: A data variable name
+        """
+
+        return key in self.data
+
+    def on_core_axis(self, var_name: str, axis_name: str) -> bool:
+        """Check core axis validation.
+
+        This function checks if a given variable loaded into a Data instance has been
+        validated on one of the core axes.
+
+        Args:
+            var_name: The name of a variable
+            axis_name: The core axis name
+
+        Returns:
+            A boolean indicating if the variable was validated on the named axis.
+
+        Raises:
+            ValueError: Unknown variable or core axis name
+            RuntimeError: Incomplete variable validation data in Data instance.
+        """
+
+        if var_name not in self.data:
+            raise ValueError(f"Unknown variable name: {var_name}")
+
+        if var_name not in self.variable_validation:
+            raise RuntimeError(f"Missing variable validation data: {var_name}")
```

Comment:
Yeah - I've switched back to `ValueError` for all of them. I'd gone for `RuntimeError` because it distinctly indicates something is wrong with the code maintaining the `variable_validation` , but the exception message is clear enough.